lib/common/actions/index.js

```diff
@@ -131,14 +132,56 @@ function getErrorMessageFromJson (
     }
     // re-assign error message after it gets used in the detail.
     errorMessage = json.message || JSON.stringify(json)
+    if (json.detail && json.detail.includes('Referenced pattern IDs:')) {
```

Comment:
You're essentially creating your own text-based protocol, which is a massive hack. This is fine for now, but we definitely need to add a comment here (including line numbers) that this needs to go and be replaced with a better solution, namely a new endpoint that returns exactly this information in JSON format

Reply:
Agreed.